go/test/endtoend/backup/vtctlbackup/backup_test.go

```diff
@@ -70,3 +81,132 @@ func setDefaultCompressionFlag() {
 	mysqlctl.ExternalDecompressorCmd = ""
 	mysqlctl.ManifestExternalDecompressorCmd = ""
 }
+
+func setDefaultCommonArgs() { commonTabletArg = getDefaultCommonArgs() }
+
+func TestBackupEngineSelector(t *testing.T) {
+	defer setDefaultCompressionFlag()
+	defer setDefaultCommonArgs()
+	defer cluster.PanicHandler(t)
+
+	// launch the custer with xtrabackup as the default engine
+	code, err := LaunchCluster(XtraBackup, "xbstream", 0, &CompressionDetails{CompressorEngineName: "pgzip"})
+	require.Nilf(t, err, "setup failed with status code %d", code)
+
+	defer TearDownCluster()
+
+	localCluster.DisableVTOrcRecoveries(t)
+	defer func() {
+		localCluster.EnableVTOrcRecoveries(t)
+	}()
+	verifyInitialReplication(t)
+
+	// first try to backup with an alternative engine (builtin)
+	err = localCluster.VtctldClientProcess.ExecuteCommand("Backup", "--allow-primary", "--backup-engine=builtin", primary.Alias)
+	require.NoError(t, err)
+	engineUsed := getBackupEngineOfLastBackup(t)
+	require.Equal(t, "builtin", engineUsed)
+
+	// then try to backup specifying the xtrabackup engine
+	err = localCluster.VtctldClientProcess.ExecuteCommand("Backup", "--allow-primary", "--backup-engine=xtrabackup", primary.Alias)
+	require.NoError(t, err)
+	engineUsed = getBackupEngineOfLastBackup(t)
+	require.Equal(t, "xtrabackup", engineUsed)
+
+	// check that by default we still use the xtrabackup engine if not specified
+	err = localCluster.VtctldClientProcess.ExecuteCommand("Backup", "--allow-primary", primary.Alias)
+	require.NoError(t, err)
+	engineUsed = getBackupEngineOfLastBackup(t)
+	require.Equal(t, "xtrabackup", engineUsed)
+}
+
+// fetch the backup engine used on the last backup triggered by the end-to-end tests.
+func getBackupEngineOfLastBackup(t *testing.T) string {
+	lastBackup := getLastBackup(t)
+
+	// open the Manifest and retrieve the backup engine that was used
+	f, err := os.Open(path.Join(localCluster.CurrentVTDATAROOT,
```

Comment:
There's an existing `readManifestFile()` function in `backup_utils.go`. Please reuse.